tools/settings.py

```diff
@@ -36,6 +37,31 @@
     'USE_PTHREADS',
     'SUPPORT_LONGJMP',
     'DEFAULT_TO_CXX',
+    'WASM_OBJECT_FILES',
+
+    # All port-related settings are valid at compile time
+    'USE_SDL',
+    'USE_LIBPNG',
+    'USE_BULLET',
+    'USE_ZLIB',
+    'USE_BZIP2',
+    'USE_VORBIS',
+    'USE_COCOS2D',
+    'USE_ICU',
+    'USE_MODPLUG',
+    'USE_SDL_MIXER',
+    'USE_SDL_IMAGE',
+    'USE_SDL_TTF',
+    'USE_SDL_NET',
+    'USE_SDL_GFX',
+    'USE_LIBJPEG',
+    'USE_OGG',
+    'USE_REGAL',
+    'USE_BOOST_HEADERS',
+    'USE_HARFBUZZ',
+    'USE_MPG123',
+    'USE_GIFLIB',
+    'USE_FREETYPE',
```

Comment:
I worry this list will get out of date when we add new ports. But offhand I don't have a better idea.

Sgtm to land this for now, but this compile-only mode warning is a little concerning. How useful is it do you think?

Reply:
I have a follow up change that means we will get hard failures if this list is no accurate.